tensorflow/core/kernels/gather_op.cc

```diff
@@ -153,6 +153,8 @@ TF_CALL_uint64(REGISTER_GATHER_CPU);
 // Registration of the GPU implementations.
 #define REGISTER_GATHER_GPU(type) REGISTER_GATHER_ALL_INDICES(GPU, type)
 
+// The following fails to compile due to lack of CudaAtomic* kernels for int64.
+//TF_CALL_int64(REGISTER_GATHER_GPU);
```

Comment:
You define int32 kernels for gather but did not register the corresponding op. So the change will have no effect.

Also, with int64 I compiled and saw no error. Not sure what was the error you saw or was it an issue with cuda version.

Reply:
I think @ebrevdo asked to remove int32 for GPU, seems like I've missed some.

I think @dantkz was the one to report this issue originally.
But if it works for you, we're probably fine now.

Reply:
Well if you removed int32 this PR seems to be purely about improving the tests for badindices.

Reply:
We can however add int64 for GPU.

Reply:
Or maybe that was only for scatter and gather, not scatter_nd and gather_nd?